src/include/daos_errno.h

```diff
@@ -31,7 +31,212 @@
+	ACTION(DER_NOREPLY,		(DER_ERR_GURT_BASE + 33))	\
+	/** denial-of-service */					\
+	ACTION(DER_DOS,			(DER_ERR_GURT_BASE + 34))       \
+	/** Incorrect target for the RPC  */				\
+	ACTION(DER_BAD_TARGET,		(DER_ERR_GURT_BASE + 35))	\
+	/** Group versioning mismatch */				\
+	ACTION(DER_GRPVER,		(DER_ERR_GURT_BASE + 36))
+	/** TODO: add more error numbers */
+
+#define D_FOREACH_DAOS_ERR(ACTION)					\
+	/** Generic I/O error */					\
+	ACTION(DER_IO,			(DER_ERR_DAOS_BASE + 1))	\
+	/** Memory free error */					\
+	ACTION(DER_FREE_MEM,		(DER_ERR_DAOS_BASE + 2))	\
+	/** Entry not found */						\
+	ACTION(DER_ENOENT,		(DER_ERR_DAOS_BASE + 3))	\
+	/** Unknown object type */					\
+	ACTION(DER_NOTYPE,		(DER_ERR_DAOS_BASE + 4))	\
+	/** Unknown object schema */					\
+	ACTION(DER_NOSCHEMA,		(DER_ERR_DAOS_BASE + 5))	\
+	/** Object is not local */					\
+	ACTION(DER_NOLOCAL,		(DER_ERR_DAOS_BASE + 6))	\
+	/** stale pool map version */					\
+	ACTION(DER_STALE,		(DER_ERR_DAOS_BASE + 7))	\
+	/** Not service leader */					\
+	ACTION(DER_NOTLEADER,		(DER_ERR_DAOS_BASE + 8))	\
+	/** * Target create error */					\
+	ACTION(DER_TGT_CREATE,		(DER_ERR_DAOS_BASE + 9))	\
+	/** Epoch is read-only */					\
+	ACTION(DER_EP_RO,		(DER_ERR_DAOS_BASE + 10))	\
+	/** Epoch is too old, all data have been recycled */		\
+	ACTION(DER_EP_OLD,		(DER_ERR_DAOS_BASE + 11))	\
+	/** Key is too large */						\
+	ACTION(DER_KEY2BIG,		(DER_ERR_DAOS_BASE + 12))	\
+	/** Record is too large */					\
+	ACTION(DER_REC2BIG,		(DER_ERR_DAOS_BASE + 13))	\
+	/** IO buffers can't match object extents */			\
+	ACTION(DER_IO_INVAL,		(DER_ERR_DAOS_BASE + 14))	\
+	/** Event queue is busy */					\
+	ACTION(DER_EQ_BUSY,		(DER_ERR_DAOS_BASE + 15))	\
+	/** Domain of cluster component can't match */			\
+	ACTION(DER_DOMAIN,		(DER_ERR_DAOS_BASE + 16))	\
+	/** Service should shut down */					\
+	ACTION(DER_SHUTDOWN,		(DER_ERR_DAOS_BASE + 17))	\
+	/** Operation now in progress */				\
+	ACTION(DER_INPROGRESS,		(DER_ERR_DAOS_BASE + 18))	\
+	/** Not applicable. */						\
+	ACTION(DER_NOTAPPLICABLE,	(DER_ERR_DAOS_BASE + 19))	\
+	/** Not a service replica */					\
+	ACTION(DER_NOTREPLICA,		(DER_ERR_DAOS_BASE + 20))	\
+	/** Checksum error */						\
+	ACTION(DER_CSUM,		(DER_ERR_DAOS_BASE + 21))	\
+	/** Unsupported durable format */				\
+	ACTION(DER_DF_INVAL,		(DER_ERR_DAOS_BASE + 22))	\
+	/** Incompatible durable format version */			\
+	ACTION(DER_DF_INCOMPT,		(DER_ERR_DAOS_BASE + 23))	\
+	/** Record size error */					\
+	ACTION(DER_REC_SIZE,		(DER_ERR_DAOS_BASE + 24))	\
+	/** Used to indicate a transaction should restart */		\
+	ACTION(DER_TX_RESTART,		(DER_ERR_DAOS_BASE + 25))	\
+	/** Data lost or not recoverable */				\
+	ACTION(DER_DATA_LOSS,		(DER_ERR_DAOS_BASE + 26))
+
+/** Defines the gurt error codes */
+#define D_FOREACH_ERR_RANGE(ACTION)	\
+	ACTION(GURT,	1000)		\
+	ACTION(DAOS,	2000)
+
+#define D_DEFINE_ERRNO(name, value) name = value,
+
+#define D_DEFINE_RANGE_ERRNO(name, base)			\
```

Comment:
(style)  Macros with multiple statements should be enclosed in a do - while loop
(style)  macros should not use a trailing semicolon